### ISSUE TITLE
Training getting killed automatically

### ISSUE THREAD

**Issue by SagarPalyal** [google-research/pegasus#148]
Hi Team,

When I am trying to fine tune pegasus model with my custom data, process is getting killed automatically when trying to save checkpoint.

**below is the error**
Instructions for updating:
Use standard file utilities to get mtimes.
W1216 12:54:09.186130 140128198473536 deprecation.py:323] From /home/ubuntu/pegasusenv/lib/python3.6/site-packages/tensorflow_core/python/training/saver.py:1069: get_checkpoint_mtimes (from tensorflow.python.training.checkpoint_management) is deprecated and will be removed in a future version.
Instructions for updating:
Use standard file utilities to get mtimes.
INFO:tensorflow:Running local_init_op.
I1216 12:54:11.458173 140128198473536 session_manager.py:500] Running local_init_op.
INFO:tensorflow:Done running local_init_op.
I1216 12:54:12.422816 140128198473536 session_manager.py:502] Done running local_init_op.
INFO:tensorflow:Saving checkpoints for 0 into /home/ubuntu/SagarPalyal/pegasusapi/models/pegasus_wiki/model.ckpt.
I1216 12:54:32.968967 140128198473536 basic_session_run_hooks.py:606] Saving checkpoints for 0 into /home/ubuntu/SagarPalyal/pegasusapi/models/pegasus_wiki/model.ckpt.
Killed

Please let me know what could be the problem as there is no error msg coming for debugging.

**Comment by agenius5** [comment]
Most probably you are running out of memory. Check your RAM and GPU usage.

**Comment by SagarPalyal** [comment]
I checked RAM is 24 percent and CPU 100 percent before training getting killed. Also I don't see any GPU utilizaiton. what should I do now to resolve this kind of issue?

**Comment by agenius5** [comment]
I'm not sure but I think your GPU must be going out of memory. Check your gpu usage with nvidia-smi on terminal. Also how much is your gpu RAM? 

And try reducing the batch size as well.

**Comment by SagarPalyal** [comment]
I can see only 110 MiB gpu memory utilization out of 15109MiB.

110MiB / 15109MiB |

And for process it is showing 99 MiB
+-----------------------------------------------------------------------------+
| Processes:                                                       GPU Memory |
|  GPU       PID   Type   Process name                             Usage      |
|=============================================================================|
|    0      6130      C   python                                        99MiB |
+-----------------------------------------------------------------------------+

Is there any way we can increase the memory utilization? I think this is the problem that finetuning is getting killed.

**Comment by agenius5** [comment]
`tf.test.is_gpu_available(
    cuda_only=False, min_cuda_compute_capability=None
)`
Run this to see if your gpu is detected.

**Comment by SagarPalyal** [comment]
I am able to train and evaluate model using my custom dataset. some ckpt files also got generated into my models directory:
model.ckpt-1000.data-00000-of-00001
model.ckpt-1000.index
model.ckpt-1000.meta

and couple of event files:
events.out.tfevents.1608192436.ip-xxx-xx-x-xx
events.out.tfevents.1608192510.ip-xxx-xx-x-xx.v2

But when I am trying running the the summarization code getting below error:
OSError: Error no file named ['pytorch_model.bin', 'tf_model.h5', 'model.ckpt.index'] found in directory /home/ubuntu/SagarPalyal/pegasusapi/models/pegasus_wiki or `from_tf` set to False

Code I am using is :

model_path ='local-pegasus-wikihow'

torch_device = 'cuda' if torch.cuda.is_available() else 'cpu'
tokenizer = PegasusTokenizer.from_pretrained(model_path)

model = PegasusForConditionalGeneration.from_pretrained(model_path).to(torch_device)

batch = tokenizer.prepare_seq2seq_batch(src_text, truncation=True, padding='longest', return_tensors="pt").to(torch_device)
translated = model.generate(**batch)
tgt_text = tokenizer.batch_decode(translated, skip_special_tokens=True)
print(tgt_text[0])


Do I need to change code here so that it will take ckpt model file for generating summary instead of searching for pre-trained pytorch_model.bin or .h5 file?

**Comment by agenius5** [comment]
Yes. I did not experiment with huggingface transformers implementation of PEGASUS. But, they use pytorch models. You need to convert the model accordingly. You can look the huggingface source code available on github to see how did they do it.
